### ISSUE TITLE
devspace upgrade downloads broken exe on windows

### ISSUE THREAD

**Issue by Andrioden** [devspace-sh/devspace#2516]
**What happened?**

```
PS C:\Projects\NHN\HN-Oppsett> devspace -v
devspace version 6.2.4

PS C:\Projects\NHN\HN-Oppsett> devspace upgrade
info Downloading newest version...
done Successfully updated to version 6.2.5
info Release note:

...
```

\->

![](https://uploads.linear.app/0aab256f-61c4-44f4-9aff-15fe09c2026b/dd4b426c-ecbc-45cc-9416-494bd91b8e44/bc2fb8ed-d77f-4607-be1b-716576144b7e)

**Local Environment:**

* DevSpace Version: 6.2.4
* Operating System: windows
* ARCH of the OS: AMD64

**Comment by lizardruss** [comment]
@Andrioden Thanks for reporting! We will take a look.

**Comment by lizardruss** [comment]
@Andrioden Can you try again? I wonder if it was just poor timing while our release process was still uploading the binary...

**Comment by meyfa** [comment]
I'm not the original poster, but I was able to reproduce this just now. The downloaded devspace.exe has only 1554 bytes, and cannot be run.

**Comment by oestoer** [comment]
manually downloading the binary from releases (e.g https://github.com/devspace-sh/devspace/releases/download/v6.2.5/devspace-windows-amd64.exe) and replacing the corrupt devspace.exe with it is a workaround until this is fixed.

**Comment by 89luca89** [comment]
Hi @Andrioden I can also reproduce the issue on Linux,
investigating on why this is happening

The file downloaded is not the updated binary but only the manifest:

```console
~$ cat devspace-linux-amd64 
{"url":"https://api.github.com/repos/devspace-sh/devspace/releases/assets/93304255","id":93304255,"node_id":"RA_kwDOCKbcz84Fj7W_","name":"devspace-linux-amd64","label":"","uploader":{"login":"github-actions[bot]","id":41898282,"node_id":"MDM6Qm90NDE4OTgyODI=","avatar_url":"https://avatars.githubusercontent.com/in/15368?v=4","gravatar_id":"","url":"https://api.github.com/users/github-actions%5Bbot%5D","html_url":"https://github.com/apps/github-actions","followers_url":"https://api.github.com/users/github-actions%5Bbot%5D/followers","following_url":"[https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}](https://api.github.com/users/github-actions%5Bbot%5D/following%7B/other_user%7D)","gists_url":"[https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}](https://api.github.com/users/github-actions%5Bbot%5D/gists%7B/gist_id%7D)","starred_url":"[https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}](https://api.github.com/users/github-actions%5Bbot%5D/starred%7B/owner%7D%7B/repo%7D)","subscriptions_url":"https://api.github.com/users/github-actions%5Bbot%5D/subscriptions","organizations_url":"https://api.github.com/users/github-actions%5Bbot%5D/orgs","repos_url":"https://api.github.com/users/github-actions%5Bbot%5D/repos","events_url":"[https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}](https://api.github.com/users/github-actions%5Bbot%5D/events%7B/privacy%7D)","received_events_url":"https://api.github.com/users/github-actions%5Bbot%5D/received_events","type":"Bot","site_admin":false},"content_type":"application/octet-stream","state":"uploaded","size":65818624,"download_count":760,"created_at":"2023-01-27T21:08:52Z","updated_at":"2023-01-27T21:08:57Z","browser_download_url":"https://github.com/devspace-sh/devspace/releases/download/v6.2.5/devspace-linux-amd64"}
```

This leads also to issue #2518 which is a duplicate of this one

**Comment by Andrioden** [comment]
@FabianKramm - At what level did you fix this? It happend again when upgrading from 6.2.5.

devspace version 6.2.5
PS C:\Projects\NHN\HN-Configuration> devspace upgrade  
info Downloading newest version...
done Successfully updated to version 6.3.0

**Comment by 89luca89** [comment]
> @FabianKramm - At what level did you fix this? It happend again when upgrading from 6.2.5.
> 
> ```
> devspace version 6.2.5
> PS C:\Projects\NHN\HN-Configuration> devspace upgrade  
> info Downloading newest version...
> done Successfully updated to version 6.3.0
> ```
> 
> -> Bugged 2kb exe file

The fix is included in 6.3.0, so from 6.2.5 to 6.3.0 a manual upgrade will be needed
